### PR TITLE
updated: load functions, merge, subset and RunDPA/RunDCA

### PR DESCRIPTION
**Pull request by ludvigla**
## Description

Currently, we have some hard coded checks to validate polarity and colocalization score tables. We can update these to be less stringent which will enable loading other spatial metrics s.a. join-count statistics and conditional probabilities. 

These checks are also used when merging and subsetting data, so we will need to update these methods as well. 

Moreover, we have set some restrictions on what metrics can be used as input for RunDPA / RunDCA. We should be less stringent here as well to enable using other metrics as input.

This PR includes the following changes:

- `ReadMPX_counts` and `ReadMPX_Seurat` now handles PXL files with new spatial metrics
- `subset` and `merge` methods for class `MPXAssay` now only requires a few columns in the spatial metrics tables. For polarity scores, columns "marker" and "component" must be present. For colocalization scores, columns "marker_1", "marker_2" and "component" must be present. 
- `RunDPA` accepts any numeric vector stored in the polarity score table as input, specified with `polarity_metric`
- `RunDCA` accepts any numeric vector stored in the colocalization score table as input, specified with `coloc_metric`

Fixes: exe-2022

## Type of change

- [x] New feature (non-breaking change which adds functionality).
- [x] This change requires a documentation update.

## How Has This Been Tested?

Currently, we do not have any example data sets with alternative spatial metrics in pixelatorR. Therefore, we only run the existing tests to make sure that the changes are backwards compatible.

## PR checklist:

- [x] This comment contains a description of changes (with reason).
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have checked my code and documentation and corrected any misspellings.
- [x] I have run R CMD check on the package and it passes without errors or warnings (notes can be acceptable if motivated)
- [x] I have documented any significant changes to the code in [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -39,9 +39,7 @@ Imports:
     pbapply,
     zip,
     jsonlite,
-    parallel,
-    lifecycle,
-    stats
+    lifecycle
```
Comment (maxkarlsson) [inline DESCRIPTION:42]: Any particular reason to remove `parallel`?

File: R/generics.R [hunk 2]
```diff
@@ -446,8 +446,8 @@ RunDPA <- function(
 #' \code{reference} will be compared to the \code{reference} group.
 #' @param reference The name of the reference group
 #' @param group_vars An optional character vector with column names to group the tests by.
-#' @param coloc_metric The colocalization metric to use. Currently, you can select one of "pearson_z" (default)
-#' or "pearson".
+#' @param coloc_metric The polarity metric to use. Any numeric data column in the colocalization score table
```
Comment (maxkarlsson) [inline R/generics.R:449]:
```suggestion
#' @param coloc_metric The colocalization metric to use. Any numeric data column in the colocalization score table
```

**Review by maxkarlsson — APPROVED** [review]
Looks good!

File: DESCRIPTION (hunk 1 above)
Comment (ludvigla) [reply DESCRIPTION:42]: Both stats and parallel are base packages and these packages doesn't have to be imported.

File: R/utils.R [hunk 3]
```diff
@@ -93,14 +93,10 @@
...
-      "'polarization' names are invalid" =
-        all(sort(names(polarization)) ==
-          sort(c(
-            "morans_i", "morans_p_value", "morans_p_adjusted",
-            "morans_z", "marker", "component"
-          )))
-    )
+    name_check <- all(c("marker", "component") %in% names(polarization))
+    if (!name_check) {
+      abort("Columns 'marker', and 'component' are required in the 'polarization' score table")
```
Comment (vincent-van-hoef) [inline R/utils.R:98]: Nice, this particular check caused some issues before when loading files from different pixelator runs. Should work now for most cases.

**Pull request merged**